test/e2e/framework/statefulset_utils.go

```diff
@@ -193,11 +194,30 @@ func getStatefulSetPodNameAtIndex(index int, ss *apps.StatefulSet) string {
 	return fmt.Sprintf("%v-%v", ss.Name, index)
 }
 
+// scaleFromStatefulSet returns a scale subresource for a statefulset.
+func scaleFromStatefulSet(ss *apps.StatefulSet, count int32) (*extensions.Scale, error) {
+	return &extensions.Scale{
+		// TODO: Create a variant of ObjectMeta type that only contains the fields below.
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ss.Name,
+			Namespace: ss.Namespace,
+		},
+		Spec: extensions.ScaleSpec{
+			Replicas: count,
+		},
+		Status: extensions.ScaleStatus{
+			Replicas: ss.Status.Replicas,
+			Selector: ss.Spec.Selector,
+		},
+	}, nil
+}
+
 // Scale scales ss to count replicas.
 func (s *StatefulSetTester) Scale(ss *apps.StatefulSet, count int32) error {
 	name := ss.Name
 	ns := ss.Namespace
-	s.update(ns, name, func(ss *apps.StatefulSet) { *(ss.Spec.Replicas) = count })
+	newScale, _ := scaleFromStatefulSet(ss, count)
+	s.update(ns, name, func(ss *apps.StatefulSet) { *(ss.Spec.Replicas) = newScale.Spec.Replicas })
```

Comment:
What you did here is:
1. create a `extensions.Scale`
1. update a StatefulSet by setting StatefulSet's `replicas` to Scale's `replicas`

This does not test `/scale` endpoint. This just sets StatefulSet's `replicas` field and is equivalent to the original code:
```go
s.update(ns, name, func(ss *apps.StatefulSet) { *(ss.Spec.Replicas) = count })
```